apps/frontend/src/components/RequestModal/RequestModal.tsx

```diff
@@ -1,47 +1,87 @@
-import { Modal } from "react-bootstrap";
-import { Form } from 'react-router-dom';
+import { Modal, Spinner } from "react-bootstrap";
 import styles from './RequestModal.module.css';
+import { RequestType, ResponseWithUser } from "../../types";
+import requestServices from "../../services/requests";
+import { useState } from "react";
+import { useRevalidator } from "react-router";
+import ResponseBox from '../ResponseBox/ResponseBox'
 
 interface Props {
   show: boolean,
-  handleClose: () => void,
+  handleCloseModal: () => void,
+  request: RequestType,
+  updateRequestList: (requests: Array<RequestType>) => void,
 }
 
-export default function RequestModal({ show, handleClose }: Props) {
-  function handleSubmit() {
-    handleClose();
-  }
+export default function RequestModal({ show, handleCloseModal, request }: Props) {
+  const [loading, setIsLoading] = useState(false);
+  const revalidator = useRevalidator();
 
-  return (
-    <Modal show={show} onHide={handleClose} animation={true} backdrop="static" centered>
-      <Modal.Header closeButton>
-        <Modal.Title>Create a request</Modal.Title>
-      </Modal.Header>
-      <Modal.Body>
-        <Form method='post' className={styles.form} name='new-request-form'>
-          <label className={styles.label} htmlFor='title'>Title:</label>
-          <input className={styles.input} type='text' name='title' id='title' required minLength={4}></input>
+  async function handleCloseRequest() {
+    try {
+      setIsLoading(true);
+      await requestServices.closeRequest(String(request.id))
+      setIsLoading(false);
+      handleCloseModal();
+      revalidator.revalidate();
```

Comment:
Why is this necessary?

Reply:
The first line calls the backend to set the request as closed.
The second line signals that the loading is done so that the loading animation can stop.
The third line simply closes the modal.
The fourth line was added because I needed a way updated the list of Requests as soon as the backend was updated. 
For example, if you have the filter set to 'Open Requests' and you select a request and close it, I want the `Open Requests` list to be updated on the spot, so that the user won't see the request he just closed anymore. 
Normally, on FSO we were taught to achieve this by holding the request list within state and update that state list. I found this approach to be very cumbersome to achieve in this particular case. (It's been a while since I've done it so I can't explain why I found it so hard).
I decided in the end that it would be easier to just re-validate the request data by calling revalidator.revalidate(). https://reactrouter.com/en/main/hooks/use-revalidator
This re-loads the data and updates the component. It is possible that my approach is slower than simply updating state since it adds and extra Request-Response cycle.

Reply:
Ok yeah this last line is what I was asking about, but using it as a last resort because useState was too complex makes sense. We could potentially look into it again in the future.